apps/web/pages/settings/my-account/profile.tsx

```diff
@@ -158,16 +158,20 @@ const ProfileView = () => {
   }>();
 
   const { reset } = formMethods;
-
+  const formInitializedRef = useRef(false);
   useEffect(() => {
-    if (user)
+    // The purpose of reset is to set the initial value obtained from tRPC.
+    // `user` would change for many reasons (e.g. when viewer.me automatically fetches on window re-focus(a react query feature))
+    if (user && !formInitializedRef.current) {
+      formInitializedRef.current = true;
       reset({
```

Comment:
Do you know why this form uses the `reset` method? Shouldn't we set it via `initialvalues` on the form component? I think that would also solve the issue, right?

Reply:
initialValues as in use form defaultValues? 
The problem here is that when useForm hook is called first time, user isn't set.

Reply:
Oh sorry, I meant `defaultValues` indeed. That also doesn't update when the `user` gets updated? I thought it did actually. 

If that doesn't work indeed, than this change is totally fine by me! 🙌

Reply:
Yeah it doesn't update. I hate this about React Hook Forms. Using them with tRPC forces us to use `reset`

<img width="908" alt="Screenshot 2022-11-01 at 1 39 46 PM" src="https://user-images.githubusercontent.com/1780212/199187909-470047cb-eaf7-498e-8082-a5cdcbff27b3.png">

Reply:
Fair! Let's go with this then instead 😄  Perhaps towards the future we could create a custom hook for this that would improve readability. Like `useSetFormValuesBasedOnData` or something like that.

Reply:
Yeah, exactly what I had in my mind.